src/runtime/containerd-shim-v2/wait.go

```diff
@@ -146,12 +149,26 @@ func watchOOMEvents(ctx context.Context, s *service) {
 				// If the GetOOMEvent call is not implemented, then the agent is most likely an older version,
 				// stop attempting to get OOM events.
 				// for rust agent, the response code is not found
-				if isGRPCErrorCode(codes.NotFound, err) {
+				if isGRPCErrorCode(codes.NotFound, err) || err.Error() == "Dead agent" {
 					return
 				}
 				continue
 			}
 
+			// write oom file for CRI-O
+			if c, ok := s.containers[containerID]; ok && oci.IsCRIOContainerManager(c.spec) {
```

Comment:
Why is this CRI-O specific?

Reply:
containerd use event to update container status, but CRI-O use a file to detect OOM event. https://github.com/cri-o/cri-o/blob/master/internal/oci/runtime_oci.go#L835-L838